scrapy/addons.py

```diff
@@ -0,0 +1,65 @@
+import logging
+from typing import TYPE_CHECKING, Any, List
+
+from scrapy.utils.conf import build_component_list
+from scrapy.utils.misc import create_instance, load_object
+
+if TYPE_CHECKING:
+    from scrapy.crawler import Crawler
+
+logger = logging.getLogger(__name__)
+
+
+class AddonManager:
```

Comment:
I wonder if we should make the methods private, at least `add`. No strong opinion, though.

Reply:
Initially it was public because users were able to add addons in their code (and do other things with them that we no longer have APIs for, like removing, disabling and changing configuration).